e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -230,25 +230,23 @@ public async Task Method_ServiceInvokeDeviceMethodWithUnknownModuleThrows()
                 actualErrorCode = ex.Code;
             }
 
-            Assert.AreEqual(ErrorCode.ModuleNotFound, actualErrorCode);
-            serviceClient.Dispose();
+            actualErrorCode.Should().Be(ErrorCode.ModuleNotFound);
         }
 
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
         public async Task Method_ServiceInvokeDeviceMethodWithNullPayload_DoesNotThrow()
         {
             // arrange
 
-            IotHubDeviceClient deviceClient = null;
             TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, "NullMethodPayloadTest").ConfigureAwait(false);
+            // TODO: switch this to MQTT after the change to MqttNet
```

Comment:
Any particular reason why? Just be we default to MQTT elsewhere now?

Reply:
Because the test fails.

Microsoft.Azure.Devices.Common.Exceptions.DeviceNotFoundException: {"Message":"{\"errorCode\":404103,\"trackingId\":\"e5800b50389c42e8abc3e4362f48cad5-G:0-TimeStamp:08/11/2022 22:45:05\",\"message\":\"The operation failed because the requested device isn't online or hasn't registered the direct method callback. To learn more, see https://aka.ms/iothub404103\",\"info\":{},\"timestampUtc\":\"2022-08-11T22:45:05.3400543Z\"}","ExceptionMessage":""}

Reply:
I'll debug it to see why it is failing.

Reply:
Fixed